### PR TITLE
Extract iceberg table

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To support the Snowflake iceberg table.

### 🤓 What?

- Update the readme to give the crawler permission to extract the iceberg tables.
- Fill extra information for iceberg table

### 🧪 Tested?

Create an iceberg table and the crawler can extract it.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: tests/mongodb/expected_datasets.json
Comment (mars-lan) [inline tests/mongodb/expected_datasets.json:1]: ?

Comment (elic-eon) [reply tests/mongodb/expected_datasets.json:1]: @usefulalgorithm Is it related to #991 ? After `pytest`, this file is changed

Comment (elic-eon) [reply tests/mongodb/expected_datasets.json:1]: Fixed, but this JSON will be modified

File: metaphor/snowflake/extractor.py [hunk 1]
```diff
@@ -709,6 +712,49 @@ def _to_dataset_eid(x: str) -> str:
...
+            return
+
+        count = 0
+        for entry in cursor:
+            (
+                table_name,
+                external_volume_name,
+                iceberg_table_type,
+            ) = (
+                entry[1],
```
Comment (mars-lan) [inline metaphor/snowflake/extractor.py:732]: I believe you can use `row[<col_nme>]` instead of index-based access?

Comment (elic-eon) [reply metaphor/snowflake/extractor.py:732]: It's tuple here, because we can't specifying columns for this query

**Review by usefulalgorithm — APPROVED** [review]

**Pull request merged**